src/config.py

```diff
@@ -0,0 +1,11 @@
+"""Configuration parser for the charm."""
+
+from pydantic import BaseModel, Field
+
+
+class CharmConfig(BaseModel):
+    """Manager for the charm configuration."""
+
+    ambient: bool
+    cni_bin_dir: str = Field(alias="cni-bin-dir", min_length=1)
+    cni_conf_dir: str = Field(alias="cni-conf-dir", min_length=1)
```

Comment:
Yeah, admittedly when implementing this I expected there to be more config options to validate :D Atm it feels heavy for just these two...  Maybe it feels better as more get added?

If we catch validation errors in `parsed_config()`, what should we do with them?  Do we have a practice in other charms for validation?  I didn't see any, but I only looked at a couple.  

atm here, setting `cni-bin-dir=""` (an invalid value) puts the charm into `Error` with a pydantic validation exception in the traceback.  That's kinda informative, but heavy-handed.  Should we do something else?

Reply:
As discussed, removed the validation requiring the string to be non-empty, since that'll fail the same as any other invalid value.  Otherwise leaving it as is